x-pack/solutions/security/plugins/security_solution/public/common/lib/search_ai_lake/hooks/integrations/use_enhanced_integration_cards.tsx

```diff
@@ -83,16 +86,24 @@ const applyCustomDisplayOrder = (
 
 export const useEnhancedIntegrationCards = (
   integrationsList: IntegrationCardItem[],
+  activeIntegrations?: GetInstalledPackagesResponse['items'],
   options?: EnhancedCardOptions
 ): { available: IntegrationCardItem[]; installed: IntegrationCardItem[] } => {
   const sorted = applyCustomDisplayOrder(integrationsList);
 
   const available = useMemo(
     () =>
       sorted.map((card) =>
-        applyCategoryBadgeAndStyling(card, IntegrationsFacets.available, options)
+        applyCategoryBadgeAndStyling(card, IntegrationsFacets.available, {
+          ...options,
+          hasDataStreams: activeIntegrations?.find(
+            (activeIntegration) => activeIntegration.name === card.name
+          )
+            ? true
+            : false,
```

Comment:
NITs: 
- We could set an empty array as default param value: 
```ts
activeIntegrations: GetInstalledPackagesResponse['items'] = []
``` 
- And here we can also avoid this ternary:

```suggestion
          hasDataStreams: activeIntegrations.some(({ name }) => name === card.name)
```